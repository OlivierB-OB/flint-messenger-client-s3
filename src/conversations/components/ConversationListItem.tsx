import ListItemAvatar from '@material-ui/core/ListItemAvatar';
import ListItemText from '@material-ui/core/ListItemText';
import Badge from '@material-ui/core/Badge';
import React, { Fragment } from 'react';
import { IConversation } from '../types';
import { ContactAvatar } from '../../users/components/ContactAvatar';
import { PrettyDate } from '../../layout/components/PrettyDate';

export interface IConversationListItemProps {
  conversation: IConversation;
}

export function ConversationListItem({ conversation }: IConversationListItemProps) {
  const { unseenMessages, messages, target, updatedAt } = conversation;
  const snippet = messages[messages.length - 1]?.content.substr(0, 25);
  return (
    <Fragment>
      <ListItemAvatar>
        <Badge badgeContent={unseenMessages} color="primary">
          <ContactAvatar target={target} />
        </Badge>
      </ListItemAvatar>
      <ListItemText
        primary={snippet}
        secondary={
          <span>
            Last update
            {': '}
            <PrettyDate date={updatedAt} />
          </span>
        }
      />
    </Fragment>
  );
}
