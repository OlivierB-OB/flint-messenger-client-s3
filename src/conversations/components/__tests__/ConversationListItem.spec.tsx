import React from 'react';
import { shallow } from 'enzyme';
import { ConversationListItem } from '../ConversationListItem';

describe('ConversationListItem', () => {
  it('should display a conversation as a list item', async () => {
    const props: any = {
      conversation: {
        unseenMessages: 'number of unseen messages',
        messages: [{ content: 'first message content' }, { content: 'last message content' }],
        target: 'conversation target',
        updatedAt: 'conversation update date',
      },
    };
    const component = shallow(<ConversationListItem {...props} />);
    expect(component).toMatchSnapshot();
  });

  it('should support the case of empty message list', async () => {
    const props: any = {
      conversation: {
        unseenMessages: 'number of unseen messages',
        messages: [],
        target: 'conversation target',
        updatedAt: 'conversation update date',
      },
    };
    const component = shallow(<ConversationListItem {...props} />);
    expect(component).toMatchSnapshot();
  });
});
