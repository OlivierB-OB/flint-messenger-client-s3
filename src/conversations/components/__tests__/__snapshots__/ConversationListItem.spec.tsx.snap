// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConversationListItem should display a conversation as a list item 1`] = `
<Fragment>
  <WithStyles(ForwardRef(ListItemAvatar))>
    <WithStyles(ForwardRef(Badge))
      badgeContent="number of unseen messages"
      color="primary"
    >
      <Connect(UserAvatar)
        target="conversation target"
      />
    </WithStyles(ForwardRef(Badge))>
  </WithStyles(ForwardRef(ListItemAvatar))>
  <WithStyles(ForwardRef(ListItemText))
    primary="last message content"
    secondary={
      <span>
        Last update
        : 
        <PrettyDate
          date="conversation update date"
        />
      </span>
    }
  />
</Fragment>
`;

exports[`ConversationListItem should support the case of empty message list 1`] = `
<Fragment>
  <WithStyles(ForwardRef(ListItemAvatar))>
    <WithStyles(ForwardRef(Badge))
      badgeContent="number of unseen messages"
      color="primary"
    >
      <Connect(UserAvatar)
        target="conversation target"
      />
    </WithStyles(ForwardRef(Badge))>
  </WithStyles(ForwardRef(ListItemAvatar))>
  <WithStyles(ForwardRef(ListItemText))
    secondary={
      <span>
        Last update
        : 
        <PrettyDate
          date="conversation update date"
        />
      </span>
    }
  />
</Fragment>
`;
